jobs/instancerunner_test.go

```diff
@@ -0,0 +1,252 @@
+	}
+
+	config["endpointTemplate"] = []string{"biz", "boz", "baz"}
+	_, err = NewInstanceRunner(config)
+	if err == nil {
+		t.Error("expected error from endpointTemplate of wrong type, got nil")
+	}
+
+	config["endpoint"] = []string{"biz", "boz", "baz"}
+	_, err = NewInstanceRunner(config)
+	if err == nil {
+		t.Error("expected error from endpoint of wrong type, got nil")
+	}
+
+	expectedEndpointTemplate := "{{.Account}}"
+	config["endpointTemplate"] = expectedEndpointTemplate
+	_, err = NewInstanceRunner(config)
+	if err != nil {
+		t.Errorf("expected nil error from set endpointTemplate, got %s", err)
+	}
+
+	expectedEndpoint := "some endpoint"
+	config["endpoint"] = expectedEndpoint
+	_, err = NewInstanceRunner(config)
+	if err != nil {
+		t.Errorf("expected nil error from set endpoint, got %s", err)
+	}
+
+	expectedToken := "my-awesome-token"
+	config["token"] = expectedToken
+
+	out, err := NewInstanceRunner(config)
+	if err != nil {
+		t.Errorf("expected nil error from set token, got %s", err)
+	}
+
+	if d := reflect.TypeOf(out).String(); d != "*jobs.InstanceRunner" {
+		t.Errorf("expected type to be '*jobs.InstanceRunner', got '%s'", d)
+	}
+
+	if out.Endpoint != expectedEndpoint {
+		t.Errorf("expected endpoint to be '%s', got '%s'", expectedEndpoint, out.Endpoint)
+	}
+
+	if out.EndpointTemplate != expectedEndpointTemplate {
+		t.Errorf("expected endpoint token to be '%s', got '%s'", expectedEndpointTemplate, out.EndpointTemplate)
+	}
+
+	if out.Token != expectedToken {
+		t.Errorf("expected token to be '%s', got '%s'", expectedToken, out.Token)
+	}
+}
+
+func TestInstanceRunnerRun(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPut {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			fmt.Fprint(w, "unexpected method "+r.Method)
+			return
+		}
+
+		tok := r.Header.Get("Auth-Token")
+		if tok != "my-awesome-token" {
+			w.WriteHeader(http.StatusForbidden)
+			fmt.Fprint(w, "bad token")
+			return
+		}
+
+		if r.URL.String() == "/" {
+			w.WriteHeader(http.StatusAccepted)
+			fmt.Fprint(w, "OK")
+		} else if r.URL.String() == "/myaccount/i-123456" {
+			w.WriteHeader(http.StatusAccepted)
+			fmt.Fprint(w, "KO")
+		} else {
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprint(w, "derp")
+		}
+	}))
+	defer ts.Close()
+
+	instanceRunner, err := NewInstanceRunner(map[string]interface{}{
+		"token":    "my-awesome-token",
+		"endpoint": ts.URL,
+	})
+	if err != nil {
+		t.Errorf("expected nil error from set token, got %s", err)
+	}
+
+	if _, err = instanceRunner.Run(context.TODO(), "", "foo"); err == nil {
+		t.Error("expected error for empty account, got nil")
+	}
+
+	if _, err = instanceRunner.Run(context.TODO(), "", map[string]string{}); err == nil {
+		t.Error("expected error for missing instance_id, got nil")
+	}
+
+	if _, err = instanceRunner.Run(context.TODO(), "", map[string]interface{}{
+		"instance_id": []string{"foo", "bar"},
```

Comment:
These tests have an empty account, so they are working (failing) but for the wrong reason, maybe we need to check the specific error message